src/core/managers/TransactionManager.ts

```diff
@@ -0,0 +1,150 @@
+import { BatchTransactionsResponseType } from 'types/serverTransactions.types';
+import { SignedTransactionType } from 'types/transactions.types';
+
+export class TransactionManager {
+  private static instance: TransactionManager | null = null;
+
+  private constructor() {}
+
+  public static getInstance(): TransactionManager {
+    if (!TransactionManager.instance) {
+      TransactionManager.instance = new TransactionManager();
+    }
+    return TransactionManager.instance;
+  }
+
+  public send = async (
+    signedTransactions: Transaction[] | Transaction[][]
+  ): Promise<string[]> => {
+    if (signedTransactions.length === 0) {
+      throw new Error('No transactions to send');
+    }
+
+    try {
+      if (this.isBatchTransaction(signedTransactions)) {
+        const sentTransactions =
+          await this.sendSignedBatchTransactions(signedTransactions);
+
+        if (!sentTransactions.data || sentTransactions.data.error) {
+          throw new Error(
+            sentTransactions.data?.error || 'Failed to send transactions'
+          );
+        }
+
+        const flatSentTransactions = this.sequentialToFlatArray(
+          sentTransactions.data.transactions
+        );
+
+        return flatSentTransactions.map((transaction) => transaction.hash);
+      }
+
+      const hashes = await this.sendSignedTransactions(signedTransactions);
+      return hashes;
+    } catch (error) {
+      const responseData = <{ message: string }>(
+        (error as AxiosError).response?.data
+      );
+      throw responseData?.message ?? (error as any).message;
+    }
+  };
+
+  private sendSignedTransactions = async (
+    signedTransactions: Transaction[]
+  ): Promise<string[]> => {
+    const { apiAddress, apiTimeout } = networkSelector(getState());
+
+    const promises = signedTransactions.map((transaction) =>
+      axios.post(`${apiAddress}/transactions`, transaction.toPlainObject(), {
+        timeout: Number(apiTimeout)
+      })
+    );
+
+    const response = await Promise.all(promises);
+
+    return response.map(({ data }) => data.txHash);
+  };
+
+  private sendSignedBatchTransactions = async (
+    signedTransactions: Transaction[][]
+  ) => {
+    const { address } = getAccount();
+    const { apiAddress, apiTimeout } = networkSelector(getState());
+
+    if (!address) {
+      return {
+        error:
+          'Invalid address provided. You need to be logged in to send transactions'
+      };
+    }
+
+    const batchId = this.buildBatchId(address);
+    const parsedTransactions = signedTransactions.map((transactions) =>
+      transactions.map((transaction) =>
+        this.parseSignedTransaction(transaction)
+      )
+    );
+
+    const payload = {
+      transactions: parsedTransactions,
+      id: batchId
+    };
+
+    const { data } = await axios.post<BatchTransactionsResponseType>(
+      `${apiAddress}/batch`,
+      payload,
+      {
+        timeout: Number(apiTimeout)
+      }
+    );
+
+    return { data };
```

Comment:
why not return data ?

Reply:
because that endpoint can return {data? SignedTxType, error?:string}